SplunkADXForwarder/app.py

```diff
@@ -0,0 +1,247 @@
+MAX_RETRIES = int(os.environ.get('MAX_RETRIES', '3'))
+MAX_SIZE = 5 * 1024 * 1024 #5MB
+DELETE_TEMP_FILE_DAYS_THRESHOLD = 1
+
+def is_file_not_being_modified(file_path):
+    """
+    Returns True if the file is currently being not modified, False otherwise.
+    """
+    file_stat = os.stat(file_path)
+    current_size = file_stat.st_size
+    time.sleep(1)  # Wait for 1 second
+    new_size = os.stat(file_path).st_size
+    if current_size != new_size:
+        logging.info("the file is currently being modified")
+    return current_size == new_size
+    
+
+def ingest_data(conn_string_builder: KustoConnectionStringBuilder, database: str, table: str, table_mapping: str) -> bool:
+    """Ingest the data into Azure Data Explorer."""
+    while True:
+         # Set the folder path to read the JSON files
+        time.sleep(60)
+        folder_path = os.getcwd()
+        buffer_folder_name = os.path.join(folder_path, "ADX_Ingest")
+        logging.info(f'ingesting message from {buffer_folder_name}')
+
+        #files = os.listdir(buffer_folder_name)
+
+        ingestion_client = QueuedIngestClient(conn_string_builder)
+        ingest_props = IngestionProperties(database=database, table=table, flush_immediately=False, data_format=DataFormat.CSV, ingestion_mapping_reference=table_mapping)
+
+        if os.path.isdir(buffer_folder_name):
+            for file in get_files_not_modified(buffer_folder_name):
+                file_path = os.path.join(buffer_folder_name, file)
+                logging.info(f"Processing file :: {file_path}")
+                if file.endswith('.csv') and is_file_not_being_modified(file_path):
+                    # ingest from file
+                    df = pd.read_csv(file_path)
+                    result = ingestion_client.ingest_from_dataframe(df, ingestion_properties=ingest_props)
+                    #if ingestion result is success or queued, then move the file to temp folder
+                    if result.status == IngestionStatus.QUEUED or result.status == IngestionStatus.SUCCESS:
+                        try:
+                            move_files_to_temp_dir(file_path)
+                            logging.info("File posted to ADX and Moved file to temp folder {}".format(file_path))
+                        except Exception as err:
+                            #if any error, continue, the file will be picked up in the next event invocation
+                            logging.error("Error while moving file to temp folder {}".format(err))
+                            continue
+                    else:
+                        logging.error("File ingestion failed, will retry ingestion in the next attempt {}".format(file_path))
+                        continue         
+        else:
+            logging.error(f"The folder {buffer_folder_name} doesnt exist")
+
+def move_files_to_temp_dir(source_file_path):
+        if not os.path.isdir(os.path.join(tempfile.gettempdir(), "ADX_Ingest")):
```

Comment:
minor lint, ADX_Ingest is a common string, can be made into a class level constant ?

Reply:
makes sense.